src/localstream.js

```diff
@@ -489,6 +595,44 @@ const localstream = (function() {
 		 */
 		close: () => {
 			close();
+		},
+		/**
+		 * Switches between cameras.
+		 */
+		switchCamera: () => {
+			_switchCamera();
+		},
+		/**
+		 * Adds the audio-and-video listener which will be called after the audio-and-video stream is established used by switching camera feature
+		 * @param cb - The audio and video listener to add 
+		 */
+		addStreamAudioVideoListener: (cb) => {
+			streamListenersAudioVideo.push(cb);
+		},
+		/**
+		 * Adds the video listener which will be called after the video stream is established used by switching camera feature
+		 * @param cb - The video listener to add 
+		 */
+		addStreamVideoListener: (cb) => {
+			streamListenersVideo.push(cb);
+		},
+        /**
+         * getting all of the video sources/cameras that are connected to the current device/browser
+         */
+		getAllVideoSources:(cb) =>{
+			_getAllVideoSources().then((sources)=>{
+				videoSources = sources;
+				cb(videoSources);
+			});
+		},
```

Comment:
Reassigning `videoSources` is not needed, it's already done within `_getAllVideoSources`.
You should prefer with a `Promise` instead of a callback. If you still want a callback, you can always do both:

```javascript
getAllVideoSources: cb => {
	return _getAllVideoSources().then(sources => {
		cb && cb(videoSources);
		return sources;
	});
}
```